rdf-abac-core/src/main/java/io/telicent/jena/abac/labels/LabelsStoreRocksDB.java

```diff
@@ -355,49 +354,51 @@ private List<String> getLabels(final ByteBuffer valueBuffer, final List<String>
      * @throws RocksDBException if something went wrong with the database lookup
      */
     private List<String> labelsForSPO(final Node subject, final Node predicate, final Node object) throws RocksDBException {
-        var labels = new ArrayList<String>();
-
-        if ( db == null ) {
+        if (db == null) {
             throw new RuntimeException("The RocksDB labels store appears to be closed.");
         }
 
-        var keySPO = keyBuffer.get().clear();
+        ByteBuffer keySPO = keyBuffer.get().clear();
+        ByteBuffer valueBuffer = this.valueBuffer.get().clear();
+
         encoder.formatTriple(keySPO, subject, predicate, object);
-        var keySP_ = keyAltBuffer.get().clear();
+        ReadOptions readOptionsInstance = readOptions.get();
+
+        var labels = new ArrayList<String>();
+        // Checking S,P,O
+        if (db.get(cfhSPO, readOptionsInstance, keySPO.flip(), valueBuffer) != RocksDB.NOT_FOUND) {
+            return getLabels(valueBuffer, labels);
+        }
+
+        ByteBuffer keySP_ = keyAltBuffer.get().clear();
```

Comment:
We only need one key buffer now the use of multiGet has been removed.

Reply:
True. It can be removed entirely from the class and presumably the same holds for the alternate value buffer too.